src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -687,6 +687,25 @@ public override void VisitExpressionStatement(ExpressionStatementSyntax node)
 
         public override void VisitLocalDeclarationStatement(LocalDeclarationStatementSyntax node)
         {
+            var invalidDimensions = ArrayBuilder<ExpressionSyntax>.GetInstance();
+            node.Declaration.Type.VisitTypeSyntaxes(type =>
+            {
+                if (type is ArrayTypeSyntax arrayType)
+                {
+                    foreach (var rankSpecifier in arrayType.RankSpecifiers)
+                    foreach (var size in rankSpecifier.Sizes)
+                    {
+                        if (size.Kind() != SyntaxKind.OmittedArraySizeExpression)
+                        {
+                            invalidDimensions.AddRange(rankSpecifier.Sizes);
+                        }
+                    }
+                }
+            });
+            foreach (var dimension in invalidDimensions)
+            {
+                Visit(dimension);
```

Comment:
>Visit(dimension); [](start = 16, length = 17)

Do we have a test that would fail if this line was removed? Please make sure we have one. It looks like SwitchExpressionSyntax needs special binder created, which wouldn't happen if we weren't visiting the sizes.

Reply:
The comment at the top of this class says that it doesn't visit expressions, so I'm not entirely sure how to test this.

> It looks like SwitchExpressionSyntax needs special binder created, which wouldn't happen if we weren't visiting the sizes.

I am not sure what you mean by this.

Reply:
> The comment at the top of this class says that it doesn't visit expressions, so I'm not entirely sure how to test this.

I think that comment was correct at some point , but not any longer. Otherwise how ```VisitSwitchExpression``` would be called, which appears to do useful things?

> I am not sure what you mean by this.

I mean that, if you incorporate a switch expression inside a size expression and do some testing with that scenario, we we are likely to run into some kind of trouble when this line (the ```Visit(dimension);```) is commented out. Could you please try that?

---
In reply to: [260143269](https://github.com/dotnet/roslyn/pull/32843#discussion_r260143269) [](ancestors = 260143269)

Reply:
Sure.
I can also raise an issue indicating that the doc comment needs looking at, if you think that's worthwhile.

Reply:
> I can also raise an issue indicating that the doc comment needs looking at, if you think that's worthwhile.

Just adjust the comment in this PR

---
In reply to: [260160197](https://github.com/dotnet/roslyn/pull/32843#discussion_r260160197) [](ancestors = 260160197)